src/transformers/utils/logging.py

```diff
@@ -30,6 +31,50 @@
 _lock = threading.Lock()
 _default_handler: Optional[logging.Handler] = None
 
+log_levels = {
+    "debug": logging.DEBUG,
+    "info": logging.INFO,
+    "warning": logging.WARNING,
+    "error": logging.ERROR,
+    "critical": logging.CRITICAL,
+}
+
+_default_log_level = logging.WARNING
+
+
+def _get_default_logging_level():
+    """
+    If TRANSFORMERS_VERBOSITY env var is set to one of the valid choices return that as the new default level.
+    If it is not - fall back to ``_default_log_level``
+    """
+    env_level_str = os.getenv("TRANSFORMERS_VERBOSITY", None)
+    if env_level_str:
+        if env_level_str in log_levels:
+            return log_levels[env_level_str]
+        else:
+            logging.getLogger().warning(
+                f"Unknown option TRANSFORMERS_VERBOSITY={env_level_str}, "
+                f"has to be one of: { ', '.join(log_levels.keys()) }"
+            )
+    return _default_log_level
+
+
+def logging_level_str_to_code(level_str):
+    """
+    Convert log level strings (error, warning, etc.) to their ``logger`` codes (``logging.ERROR``, ``logging.WARNING``, etc.)
+
+    Args:
+        level_str (:obj:`str`): "error", "warning", etc.
+
+    Returns:
+        :obj:`int`: The corresponding logging level
+    """
+
+    if level_str in log_levels:
+        return log_levels[level_str]
+    else:
+        raise ValueError(f"unknown level {level_str}, has to be one of: { log_levels.keys() }")
```

Comment:
Is this only meant to be used in tests?

Reply:
I wasn't sure whether to make it a private function (or move it elsewhere). My thinking was that if we provide `set_verbosity` with decimal codes (10, 20), there should be a way to set it with strings, e.g. if someone writes an application with cli. e.g. as in the proposed `pytest` flag https://github.com/huggingface/transformers/pull/6816

I'm totally fine to remove it  for now and just lookup `logging.log_levels[level_str]` directly in the test.

Reply:
I think we can remove this method and just lookup `logging.log_levels[level_str]` in the tests